src/actions/auth/registrationAction.js

```diff
@@ -0,0 +1,61 @@
+import axios from 'axios';
+import { toast } from 'react-toastify';
+import RegistrationConstants from './actionTypes';
+
+export function registerUserSuccess(newUser) {
+  return {
+    type: RegistrationConstants.REGISTER_SUCCESS,
+    payload: newUser,
+  };
+}
+
+export function registerUserFail(errorMessage) {
+  return {
+    type: RegistrationConstants.REGISTER_FAILURE,
+    payload: errorMessage,
+  };
+}
+
+export function registerRequest(newUser) {
+  return {
+    type: RegistrationConstants.REGISTER_REQUEST,
+    payload: newUser,
+  };
+}
+
+export function registerUser(newUser) {
+  // eslint-disable-next-line func-names
+  return function (dispatch) {
+    dispatch(registerRequest(newUser));
+    return axios.post('https://ah-backend-kronos-staging.herokuapp.com/api/users/', newUser, {
+      headers: { 'Content-Type': 'application/json' },
+    })
+      .then((response) => {
+        dispatch(registerUserSuccess(response.data));
+        toast.success('Account created. Check your email to verify', {
+          containerId: 'A',
+        });
+        setTimeout(() => {
+          window.location.replace('/');
+        }, 1000);
+      })
+      .catch((error) => {
+        dispatch(registerUserFail(error.response.data));
+        const errors = error.response.data;
+        if (Object.keys(errors.errors).length === 1) {
+          toast.error(Object.values(errors.errors)[0][0], {
+            containerId: 'A',
+          });
+        } else {
+          toast.error('Email and username already exist', {
```

Comment:
We can agreed you were going to create functions for showing toast success and toast error messages

Reply:
Thanks this is also resolved.